sdk/storage/azure-storage-files-datalake/src/datalake_file_system_client.cpp

```diff
@@ -194,11 +195,32 @@ namespace Azure { namespace Storage { namespace Files { namespace DataLake {
     Models::CreateFileSystemResult ret;
     ret.ETag = std::move(result->ETag);
     ret.LastModified = std::move(result->LastModified);
+    ret.Created = true;
     return Azure::Core::Response<Models::CreateFileSystemResult>(
         std::move(ret), result.ExtractRawResponse());
   }
 
-  Azure::Core::Response<Models::FileSystemDeleteResult> FileSystemClient::Delete(
+  Azure::Core::Response<Models::CreateFileSystemResult> FileSystemClient::CreateIfNotExists(
+      const CreateFileSystemOptions& options) const
+  {
+    try
+    {
+      return Create(options);
+    }
+    catch (StorageException& e)
+    {
+      if (e.ErrorCode == Details::ContainerAlreadyExists)
+      {
+        Models::CreateFileSystemResult ret;
+        ret.Created = false;
+        return Azure::Core::Response<Models::CreateFileSystemResult>(
+            std::move(ret), std::move(e.RawResponse));
+      }
+      throw;
+    }
+  }
```

Comment:
why not call `BlobContainerClient::CreateIfNotExists`?

Reply:
It does not have `Created` in the response.

Reply:
I will add that, but I don't think that will affect your impl here. You'll need to instantiate a `DeleteFilesystemResult` anyway.